### PR TITLE
ToTokens refactor

### PR DESCRIPTION
**Pull request by cataggar**
This is a good milestone for the `ToTokens` refactor. Several new types are created that implement [quote::ToTokens](https://docs.rs/quote/latest/quote/trait.ToTokens.html). This assists in testing the transform from OpenAPI types. Any types that implement `ToTokens` can be used directing in `quote!` macro. This makes it possible to incrementally refactor out the types.

### CODE REVIEW

File: services/autorust/codegen/src/codegen_models.rs [hunk 1]
```diff
@@ -703,40 +708,65 @@ fn create_struct(cg: &CodeGen, schema: &SchemaGen, struct_name: &str, pageable:
     Ok(code)
 }
 
-struct TypeCode {
-    type_name: TokenStream,
-    code: TokenStream,
+struct FieldCode {
```
Comment (rylev) [inline services/autorust/codegen/src/codegen_models.rs:711]: nit: maybe calling this `StructFieldCode`?

File: services/autorust/codegen/src/codegen_models.rs [hunk 2]
```diff
@@ -703,40 +708,65 @@ fn create_struct(cg: &CodeGen, schema: &SchemaGen, struct_name: &str, pageable:
...
+
+impl ToTokens for FieldCode {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        if let Some(code) = &self.code {
+            code.to_tokens(tokens)
+        }
+    }
+}
+
+enum TypeCode {
```
Comment (rylev) [inline services/autorust/codegen/src/codegen_models.rs:724]: For all these "code" types, I'd really appreciate a doc comment that points out what they are:

```rust
/// A struct field
///
/// ```rust
/// struct My Struct {
///   my_field: u32 // <-- refers to this
/// }
/// ```
struct FieldCode {
```

**Review by rylev — APPROVED** [review]
Looks good. I have some small suggestions for improvement, but nothing blocking. 🎉

File: services/autorust/codegen/src/codegen.rs [hunk 3]
```diff
@@ -199,6 +178,258 @@ pub fn parse_params(path: &str) -> Vec<String> {
...
+pub struct TypeNameCode {
+    type_path: TypePath,
+    should_force_obj: bool,
+    is_option: bool,
+    vec_count: i32,
+    add_into: bool,
+    allow_add_into: bool,
+    add_box: bool,
+    qualify_models: bool,
+    allow_qualify_models: bool,
```
[the remark this answers is not in the document]
Comment (cataggar) [reply services/autorust/codegen/src/codegen.rs]: I made the suggested name changes.

**Pull request merged**